org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/cvc_complex_type_2_4_bCodeAction.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.extensions.contentmodel.model.CMDocument;
+import org.eclipse.lemminx.extensions.contentmodel.model.CMElementDeclaration;
+import org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager;
+import org.eclipse.lemminx.extensions.contentmodel.utils.XMLGenerator;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionRequest;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+/**
+ * CodeAction to insert expected child elements within an element
+ * 
+ * Given this XML where the expected child elements are not present as defined
+ * in the XSD:
+ * 
+ * <servlet></servlet>
+ * Error:
+ * Child elements are missing from element:
+ * - servlet
+ *
+ * The following elements are expected:
+ * - description
+ * - display-name
+ * - icon
+ * - servlet-name
+ * 
+ * To fix the error, the code action will suggest inserting the expected
+ * elements inside the parent tag
+ * 
+ */
+
+public class cvc_complex_type_2_4_bCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeActions, CancelChecker cancelChecker) {
+		DOMDocument document = request.getDocument();
+		Diagnostic diagnostic = request.getDiagnostic();
+		try {
+			Range diagnosticRange = diagnostic.getRange();
+			int startOffset = document.offsetAt(diagnosticRange.getStart()) + 1;
+			DOMNode node = document.findNodeAt(startOffset);
+
+			if (node == null || !node.isElement()) {
+				return;
+			}
+
+			DOMElement element = (DOMElement) node;
+			int elementOffset = element.getEndTagOpenOffset();
```

Comment:
Here you get the end tag open offset:

```xml
<foo>

|</foo>
```

But it is wrong. Imagine you have this XML:


```xml
<foo>

                         |</foo>
```

It will not insert to the correct position.

I think the correct position is 

`int elementOffset = element.getStartTagCloseOffset();`

```xml
<foo>|

</foo>
```

By using this offset, you should generate the `\r\n`  without problem I think (I hope).